internal/server/handlers/rbac/monitoring_instance_test.go

```diff
@@ -0,0 +1,743 @@
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "read-only for 'inaccessible-cfg' in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, read, default/inaccessible-cfg",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "read-only for all monitoring-instances in kube-system namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, read, kube-system/*",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+		}
+
+		ctx := context.WithValue(context.Background(), common.UserCtxKey, "bob")
+		for _, tc := range testCases {
+			t.Run(tc.desc, func(t *testing.T) {
+				t.Parallel()
+				k8sMock := newConfigMapMock(tc.policy)
+				enf, err := rbac.NewEnforcer(ctx, k8sMock, zap.NewNop().Sugar())
+				require.NoError(t, err)
+
+				next := next()
+
+				h := &rbacHandler{
+					next:       next,
+					log:        zap.NewNop().Sugar(),
+					enforcer:   enf,
+					userGetter: testUserGetter,
+				}
+				_, err = h.GetMonitoringInstance(ctx, "default", "monitoring-instance-1")
+				assert.ErrorIs(t, err, tc.wantErr)
+			})
+		}
+	})
+
+	t.Run("CreateMonitoringInstance", func(t *testing.T) {
+		t.Parallel()
+		next := func() *handlers.MockHandler {
+			next := handlers.MockHandler{}
+			next.On("CreateMonitoringInstance", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(
+				&everestv1alpha1.MonitoringConfig{}, nil,
+			)
+			return &next
+		}
+
+		testCases := []struct {
+			desc    string
+			policy  string
+			wantErr error
+		}{
+			{
+				desc: "admin",
+				policy: newPolicy(
+					"g, bob, role:admin",
+				),
+			},
+			{
+				desc: "all actions for all monitoring-instances in all namespaces",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, */*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "all actions for all monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, default/*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "all actions for 'monitoring-instance-1' in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, default/monitoring-instance-1",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for all monitoring-instances in all namespaces",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, */*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for all monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, default/*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for 'monitoring-instance-1' in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, default/monitoring-instance-1",
+					"g, bob, role:test",
+				),
+			},
```

Comment:
Should we have a negative case for the resource name enforcement?
```suggestion
			},			{
				desc: "create only action for 'some' in default namespace",
				policy: newPolicy(
					"p, role:test, monitoring-instances, create, default/some",
					"g, bob, role:test",
				),
				wantErr: ErrInsufficientPermissions,
			},
```